packages/core/js-client/src/services/builtins.ts

```diff
@@ -14,74 +14,142 @@
 import { Buffer } from "buffer";
 
 import { JSONValue } from "@fluencelabs/interfaces";
 import bs58 from "bs58";
 import { sha256 } from "multiformats/hashes/sha2";
+import { z } from "zod";
 
 import {
+  CallServiceData,
   CallServiceResult,
   CallServiceResultType,
   GenericCallServiceHandler,
   ResultCodes,
 } from "../jsServiceHost/interfaces.js";
-import { getErrorMessage, isString, jsonify } from "../util/utils.js";
+import { getErrorMessage, jsonify } from "../util/utils.js";
 
-const success = (
-  // TODO: Remove unknown after adding validation to builtin inputs
-  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
-  result: CallServiceResultType | unknown,
-): CallServiceResult => {
+const success = (result: CallServiceResultType): CallServiceResult => {
   return {
-    // TODO: Remove type assertion after adding validation to builtin inputs
-    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-    result: result as CallServiceResultType,
+    result,
     retCode: ResultCodes.success,
   };
 };
 
-const error = (
-  // TODO: Remove unknown after adding validation to builtin inputs
-  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
-  error: CallServiceResultType | unknown,
-): CallServiceResult => {
+const error = (error: CallServiceResultType): CallServiceResult => {
   return {
-    // TODO: Remove type assertion after adding validation to builtin inputs
-    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-    result: error as CallServiceResultType,
+    result: error,
     retCode: ResultCodes.error,
   };
 };
 
+const chunk = <T>(arr: T[]): T[][] => {
+  const res: T[][] = [];
+  const chunkSize = 2;
+
+  for (let i = 0; i < arr.length; i += chunkSize) {
+    const chunk = arr.slice(i, i + chunkSize);
+    res.push(chunk);
+  }
+
+  return res;
+};
+
 const errorNotImpl = (methodName: string) => {
   return error(
     `The JS implementation of Peer does not support "${methodName}"`,
   );
 };
 
-const makeJsonImpl = (args: [Record<string, JSONValue>, ...JSONValue[]]) => {
-  const [obj, ...kvs] = args;
+const parseWithSchema = <T extends z.ZodTypeAny>(
+  schema: T,
+  req: CallServiceData,
+): [z.infer<T>, null] | [null, string] => {
+  const result = schema.safeParse(req.args, {
+    errorMap: (issue, ctx) => {
+      if (issue.code === z.ZodIssueCode.invalid_type) {
+        if (issue.path.length === 1 && typeof issue.path[0] === "number") {
+          const [arg] = issue.path;
+          return {
+            message: `Argument ${arg} expected to be of type ${issue.expected}, Got ${issue.received}`,
+          };
+        }
+      }
+
+      if (issue.code === z.ZodIssueCode.too_big) {
+        return {
+          message: `Expected ${
+            issue.maximum
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          } argument(s). Got ${ctx.data.length}`,
```

Comment:
I suggest to have a variable where you access this any and disable esling only once

Reply:
Not possible to do this somewhere above these conditions. This prop available only when condition is met